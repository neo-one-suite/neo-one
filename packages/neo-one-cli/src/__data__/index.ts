export * from './getClients';
export * from './getContracts';
export * from './buildTests';
export * from './generateTests';
