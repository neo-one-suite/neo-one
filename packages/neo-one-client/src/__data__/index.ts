export * from './keys';
export * from './transactions';
