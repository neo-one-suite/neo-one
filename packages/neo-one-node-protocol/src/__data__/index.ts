export { context } from './context';
export { keys } from './keys';
