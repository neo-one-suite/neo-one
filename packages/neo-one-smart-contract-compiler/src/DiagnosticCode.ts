export enum DiagnosticCode {
  UNSUPPORTED_SYNTAX = 0,
  UNKNOWN_TYPE,
  EXPECTED_NUMBER,
  EXPECTED_STRING,
  REFERENCE_ERROR,
  SOMETHING_WENT_WRONG,
  EXPECTED_PROPERTY,
  EXPECTED_CAST,
  TRANSPILATION_ERROR,
  INVALID_CONTRACT_PARAMETER,
  INVALID_CONTRACT_TYPE,
  INVALID_CONTRACT_METHOD,
  INVALID_CONTRACT_EVENT,
  INVALID_VALUE_TYPE,
  INVALID_SYS_CALL,
  UNKNOWN_SYMBOL,
}
