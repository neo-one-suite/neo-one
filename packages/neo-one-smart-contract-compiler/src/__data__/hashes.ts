export const hashes = {
  OLD_NEO_ASSET_HASH: '0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b',
  OLD_GAS_ASSET_HASH: '0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7',
};
