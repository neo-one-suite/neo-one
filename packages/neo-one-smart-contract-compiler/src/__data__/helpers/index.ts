export * from './execute';
export * from './extractors';
export * from './startNode';
