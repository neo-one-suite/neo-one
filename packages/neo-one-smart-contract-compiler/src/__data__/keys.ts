import { common } from '@neo-one/client-core';

export const keys = [
  {
    name: 'first',
    address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('7d128a6d096f0c14c3a25a2b0c41cf79661bfcb4a8cc95aaaea28bde4d732344', 'hex'),
    ),
    publicKey: common.bufferToECPoint(
      Buffer.from('02028a99826edc0c97d18e22b6932373d908d323aa7f92656a77ec26e8861699ef', 'hex'),
    ),
    wif: 'L1QqQJnpBwbsPGAuutuzPTac8piqvbR1HRjrY5qHup48TBCBFe4g',
    password: 'city of zion',
    encryptedWIF: '6PYLHmDf6AjF4AsVtosmxHuPYeuyJL3SLuw7J1U8i7HxKAnYNsp61HYRfF',
    scriptHash: common.bufferToUInt160(Buffer.from('3775292229eccdf904f16fff8e83e7cffdc0f0ce', 'hex')),
  },
  {
    name: 'second',
    address: 'ALfnhLg7rUyL6Jr98bzzoxz5J7m64fbR4s',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('9ab7e154840daca3a2efadaf0df93cd3a5b51768c632f5433f86909d9b994a69', 'hex'),
    ),
    publicKey: common.bufferToECPoint(
      Buffer.from('031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c9', 'hex'),
    ),
    wif: 'L2QTooFoDFyRFTxmtiVHt5CfsXfVnexdbENGDkkrrgTTryiLsPMG',
    password: '我的密码',
    encryptedWIF: '6PYWVp3xfgvnuNKP7ZavSViYvvim2zuzx9Q33vuWZr8aURiKeJ6Zm7BfPQ',
    scriptHash: common.bufferToUInt160(Buffer.from('35b20010db73bf86371075ddfba4e6596f1ff35d', 'hex')),
  },
  {
    name: 'third',
    address: 'AVf4UGKevVrMR1j3UkPsuoYKSC4ocoAkKx',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('3edee7036b8fd9cef91de47386b191dd76db2888a553e7736bb02808932a915b', 'hex'),
    ),
    publicKey: common.bufferToECPoint(
      Buffer.from('02232ce8d2e2063dce0451131851d47421bfc4fc1da4db116fca5302c0756462fa', 'hex'),
    ),
    wif: 'KyKvWLZsNwBJx5j9nurHYRwhYfdQUu9tTEDsLCUHDbYBL8cHxMiG',
    password: 'MyL33tP@33w0rd',
    encryptedWIF: '6PYNoc1EG5J38MTqGN9Anphfdd6UwbS4cpFCzHhrkSKBBbV1qkbJJZQnkn',
    scriptHash: common.bufferToUInt160(Buffer.from('9847e26135152874355e324afd5cc99f002acb33', 'hex')),
  },
];
