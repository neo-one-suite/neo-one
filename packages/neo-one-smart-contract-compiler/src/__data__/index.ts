import * as helpers from './helpers';

export { helpers };
export * from './keys';
