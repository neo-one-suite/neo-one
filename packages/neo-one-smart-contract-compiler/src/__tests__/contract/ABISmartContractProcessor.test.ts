import { helpers } from '../../__data__';
import { DiagnosticCode } from '../../DiagnosticCode';

const properties = `
public readonly properties = {
  trusts: '*',
  permissions: [],
  groups: [],
};
`;

describe('ABISmartContractProcessor', () => {
  test('no duplicate events', async () => {
    await helpers.compileString(
      `
      import { SmartContract, createEventNotifier } from '@neo-one/smart-contract';

      const first = createEventNotifier('foo');
      const second = createEventNotifier<number>('foo', 'arg');

      first();
      second(1);

      export class TestSmartContract extends SmartContract {
        ${properties}
      }
    `,
      { type: 'error' },
    );
  });

  test('invalid method parameter', async () => {
    await helpers.compileString(
      `
      import { SmartContract } from '@neo-one/smart-contract';

      class Foo { bar(): string { return ''; } }

      export class TestSmartContract extends SmartContract {
        ${properties}

        public test(value: Foo): void {
          // do nothing
        }
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidContractType },
    );
  });

  test('invalid method return', async () => {
    await helpers.compileString(
      `
      import { SmartContract } from '@neo-one/smart-contract';

      class Foo { bar(): string { return ''; } }

      export class TestSmartContract extends SmartContract {
        ${properties}

        public test(): Foo {
          // do nothing
        }
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidContractType },
    );
  });

  test('invalid method return - union', async () => {
    await helpers.compileString(
      `
      import { SmartContract } from '@neo-one/smart-contract';

      export class TestSmartContract extends SmartContract {
        ${properties}
        private foo: boolean = true;

        public test() {
          if (this.foo) {
            return '';
          }

          return 0;
        }
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidContractType },
    );
  });

  test('invalid getter', async () => {
    await helpers.compileString(
      `
      import { SmartContract } from '@neo-one/smart-contract';

      class Foo { bar(): string { return ''; } }

      export class TestSmartContract extends SmartContract {
        ${properties}

        public get foo(): Foo {
          return new Foo();
        }
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidContractType },
    );
  });

  test('invalid setter', async () => {
    await helpers.compileString(
      `
      import { SmartContract } from '@neo-one/smart-contract';

      class Foo {
        doSomething() {
          // do nothing
        }
      }

      export class TestSmartContract extends SmartContract {
        ${properties}

        public set foo(value: Foo): void {
          value.doSomething();
        }
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidContractType },
    );
  });

  test('invalid getter + setter', async () => {
    await helpers.compileString(
      `
      import { SmartContract } from '@neo-one/smart-contract';

      class Foo {
        doSomething() {
          // do nothing
        }
      }

      export class TestSmartContract extends SmartContract {
        ${properties}

        public get foo(): Foo {
          return new Foo();
        }

        public set foo(value: Foo): void {
          value.doSomething();
        }
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidContractType },
    );
  });
});
