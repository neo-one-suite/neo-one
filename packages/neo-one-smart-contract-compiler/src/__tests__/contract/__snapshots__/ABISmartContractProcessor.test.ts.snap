// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ABISmartContractProcessor invalid getter + setter 1`] = `
"snippetCode.ts (5,9): Invalid SmartContract type. See https://neo-one.io/docs/methods#Parameter-and-Return-Types for valid parameter and return types.

      3 | 
      4 |       class Foo {
    > 5 |         doSomething() {
        |         ^
      6 |           // do nothing
      7 |         }
      8 |       }
"
`;

exports[`ABISmartContractProcessor invalid getter 1`] = `
"snippetCode.ts (4,19): Invalid SmartContract type. See https://neo-one.io/docs/methods#Parameter-and-Return-Types for valid parameter and return types.

      2 |       import { SmartContract } from '@neo-one/smart-contract';
      3 | 
    > 4 |       class Foo { bar(): string { return ''; } }
        |                   ^
      5 | 
      6 |       export class TestSmartContract extends SmartContract {
      7 |         
"
`;

exports[`ABISmartContractProcessor invalid method parameter 1`] = `
"snippetCode.ts (4,19): Invalid SmartContract type. See https://neo-one.io/docs/methods#Parameter-and-Return-Types for valid parameter and return types.

      2 |       import { SmartContract } from '@neo-one/smart-contract';
      3 | 
    > 4 |       class Foo { bar(): string { return ''; } }
        |                   ^
      5 | 
      6 |       export class TestSmartContract extends SmartContract {
      7 |         
"
`;

exports[`ABISmartContractProcessor invalid method return - union 1`] = `
"snippetCode.ts (14,9): Invalid SmartContract type. See https://neo-one.io/docs/methods#Parameter-and-Return-Types for valid parameter and return types.

      12 |         private foo: boolean = true;
      13 | 
    > 14 |         public test() {
         |         ^
      15 |           if (this.foo) {
      16 |             return '';
      17 |           }
"
`;

exports[`ABISmartContractProcessor invalid method return 1`] = `
"snippetCode.ts (4,19): Invalid SmartContract type. See https://neo-one.io/docs/methods#Parameter-and-Return-Types for valid parameter and return types.

      2 |       import { SmartContract } from '@neo-one/smart-contract';
      3 | 
    > 4 |       class Foo { bar(): string { return ''; } }
        |                   ^
      5 | 
      6 |       export class TestSmartContract extends SmartContract {
      7 |         
"
`;

exports[`ABISmartContractProcessor invalid setter 1`] = `
"snippetCode.ts (5,9): Invalid SmartContract type. See https://neo-one.io/docs/methods#Parameter-and-Return-Types for valid parameter and return types.

      3 | 
      4 |       class Foo {
    > 5 |         doSomething() {
        |         ^
      6 |           // do nothing
      7 |         }
      8 |       }
"
`;

exports[`ABISmartContractProcessor no duplicate events 1`] = `
"snippetCode.ts (5,50): Duplicate event. Only one event is allowed per name.

      3 | 
      4 |       const first = createEventNotifier('foo');
    > 5 |       const second = createEventNotifier<number>('foo', 'arg');
        |                                                  ^
      6 | 
      7 |       first();
      8 |       second(1);
"
`;
